components/mosart/src/riverroute/MOSART_RES_type.F90

```diff
@@ -0,0 +1,321 @@
+
+    integer :: ier                  ! error code
+    integer :: begr, endr, iunit, nn, n, cnt, nr, nt
+    integer  :: damID
+    character(len=*),parameter :: subname = '(MOSART_reservoir_init)'
+    character(len=*),parameter :: FORMI = '(2A,2i10)'
+    character(len=*),parameter :: FORMR = '(2A,2g15.7)'
+    type(file_desc_t):: ncid       ! netcdf file
+    type(var_desc_t) :: vardesc    ! netCDF variable description
+    type(io_desc_t)    :: iodesc_dbl ! pio io desc
+
+    begr = rtmCTL%begr
+    endr = rtmCTL%endr  
+
+    if(endr >= begr) then
+        allocate(Tres_para%Tres(begr:endr))
+        Tres_para%Tres = 0._r8
+        allocate(Tres_para%Eff_trapping(begr:endr))
+        Tres_para%Eff_trapping = 0._r8
+        do iunit=begr,endr
+            damID = WRMUnit%INVicell(iunit) 
+            if (.not.(damID > ctlSubwWRM%LocalNumDam .OR. damID <= 0 .or. WRMUnit%MeanMthFlow(damID,13) <= 0.01_r8)) then
+                Tres_para%Tres(iunit) = CRTres(WRMUnit%StorCap(damID), WRMUnit%MeanMthFlow(damID,13))
+                Tres_para%Eff_trapping(iunit) = CREff_trapping(Tres_para%Tres(iunit))
+                !write(iulog,*) ' Reservoir Trapping ', iunit, WRMUnit%StorCap(damID), WRMUnit%MeanMthFlow(damID,13), Tres_para%Tres(iunit), Tres_para%Eff_trapping(iunit)
+            else
+                Tres_para%Tres(iunit) = 0._r8
+                Tres_para%Eff_trapping(iunit) = 0._r8
+            end if
+        end do    
+
+        allocate (Tres%wres(begr:endr,nt_rtm))
+        Tres%wres = 0._r8
+
+        allocate (Tres%dwres(begr:endr,nt_rtm))
+        Tres%dwres = 0._r8
+
+        allocate (Tres%eres_in(begr:endr,nt_rtm))
+        Tres%eres_in = 0._r8
+
+        allocate (Tres%eres_out(begr:endr,nt_rtm))
+        Tres%eres_out = 0._r8    
+
+if(0>1) then ! comment out temporily, please do not delete
+        allocate(Tres_para%Eff_trapping_t(begr:endr))
+        Tres_para%Eff_trapping_t = 0._r8
+        do iunit=begr,endr
+		    if(Tres_para%Tres_t(iunit)>0._r8) then
+                Tres_para%Eff_trapping_t(iunit) = CREff_trapping(Tres_para%Tres_t(iunit))
+			else
+                Tres_para%Eff_trapping_t(iunit) = 0._r8
+			end if
+		end do
+
+        allocate(Tres_para%Eff_trapping_r(begr:endr))
+        Tres_para%Eff_trapping_r = 0._r8
+        do iunit=begr,endr
+		    if(Tres_para%Tres_r(iunit)>0._r8) then
+               Tres_para%Eff_trapping_r(iunit) = CREff_trapping(Tres_para%Tres_r(iunit))
+			else
+                Tres_para%Eff_trapping_r(iunit) = 0._r8
+			end if
+		end do
+end if
+
+        allocate (Tres%wres_t(begr:endr,nt_rtm))
+        Tres%wres_t = 0._r8
+
+        allocate (Tres%dwres_t(begr:endr,nt_rtm))
+        Tres%dwres_t = 0._r8
+
+        allocate (Tres%eres_in_t(begr:endr,nt_rtm))
+        Tres%eres_in_t = 0._r8
+
+        allocate (Tres%eres_out_t(begr:endr,nt_rtm))
+        Tres%eres_out_t = 0._r8    
+
+    end if  
+
+  end subroutine MOSART_reservoir_init  
+
+
+    subroutine MOSART_reservoir_init_old
```

Comment:
Look like MOSART_reservoir_init_old is legacy code and should be removed.

Reply:
Deleted